api/revisions/serializers.py

```diff
@@ -0,0 +1,156 @@
+from api.base.serializers import (
+    JSONAPISerializer,
+    LinksField,
+    RelationshipField,
+    VersionedDateTimeField,
+)
+
+from rest_framework import serializers as ser
+from api.base.utils import absolute_reverse
+from osf.models.schema_responses import SchemaResponses, SchemaResponseBlock
+from osf.models import Registration, RegistrationSchemaBlock
+from rest_framework import exceptions
+from django.utils import timezone
+from api.base.serializers import JSONAPISerializer, LinksField
+from django.contrib.contenttypes.models import ContentType
+
+
+class SchemaResponsesSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'date_created',
+        'date_modified',
+        'revision_justification',
+        'reviews_state',
+    ])
+
+    writeable_method_fields = frozenset([
+        'revision_response',
+    ])
+
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    date_created = VersionedDateTimeField(source='created', required=False)
+    date_modified = VersionedDateTimeField(source='modified', required=False)
+    revision_justification = ser.CharField(required=False)
+    revision_response = ser.JSONField(source='schema_responses', required=False)
+    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], required=False)
+    is_pending_current_user_approval = ser.SerializerMethodField()
+
+    links = LinksField(
+        {
+            'self': 'get_absolute_url',
+        },
+    )
+
+    registration = RelationshipField(
+        related_view='registrations:registration-detail',
+        related_view_kwargs={'node_id': '<parent._id>'},
+        required=False,
+    )
+
+    registration_schema = RelationshipField(
+        related_view='schemas:registration-schema-detail',
+        related_view_kwargs={'schema_id': '<parent.registered_schema_id>'},
+        read_only=True,
+        required=False,
+    )
+
+    initiated_by = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<initiator._id>'},
+        read_only=True,
+        required=False,
+
+    )
+
+    class Meta:
+        type_ = 'revisions'
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'revisions:schema-responses-detail',
+            kwargs={
+                'version': self.context['request'].parser_context['kwargs']['version'],
+                'revision_id': obj._id,
+            },
+        )
+
+    def get_is_pending_current_user_approval(self, obj):
+        # TBD
+        return False
+
+
+class SchemaResponsesListSerializer(SchemaResponsesSerializer):
+
+    # overrides Serializer
+    def is_valid(self, clean_html=True, **kwargs):
```

Comment:
I'm not sure why this is necessary either, but it won't pick up these fields unless I do this.